File: algorithms/problem_0276/leetcode3.py
class Solution:
    def numWays(self, n: int, k: int) -> int:
        if n == 0:
            return 0
        same, different = 0, k
        for i in range(2, n + 1):
            same, different = different, (different + same) * (k - 1)
        return same + different

File: algorithms/problem_0276/LeetCode3.java
class Solution {
    public int numWays(int n, int k) {
        if (n == 0) {
            return 0;
        }
        int same = 0;
        int different = k;
        for (int i = 2; i <= n; i++) {
            int tmp = same;
            same = different;
            different = (tmp + different) * (k - 1);
        }
        return same + different;
    }
}
